### PR TITLE
batches: add rename notice and update changelog

### PR DESCRIPTION
**Pull request by LawnGnome**
I think I got the capitalisation right on the places we use "batch change" in here, but this is all new and scary.

Further suggestions for the changelog welcome. The burndown chart wording is pretty rough.

### CODE REVIEW

File: client/web/src/enterprise/batches/list/BatchChangesListIntro.tsx [hunk 1]
```diff
@@ -36,18 +43,44 @@ const BatchChangesChangelogAlert: React.FunctionComponent = () => (
...
+        className="batch-changes-list-intro__alert pb-4"
+        partialStorageKey="batch-changes-list-intro-rename"
+    >
+        <div className="batch-changes-list-intro__card card h-100 p-2">
+            <div className="card-body">
+                <h4>Campaigns is now Batch Changes</h4>
+                <p className="text-muted mb-0">
+                    Campaigns was renamed to Sourcegraph Batch Changes in version 3.26. If you were already using it
+                    under the previous name (campaigns), backwards compatibility has been preserved.{' '}
+                    <a href="https://docs.sourcegraph.com/batch_changes/references/name-change">Read more.</a>
```
Comment (eseliger) [inline client/web/src/enterprise/batches/list/BatchChangesListIntro.tsx:77]: underscore and hyphen in the same url 😨

File: client/web/src/enterprise/batches/list/BatchChangesListIntro.tsx [hunk 2]
```diff
@@ -36,18 +43,44 @@ const BatchChangesChangelogAlert: React.FunctionComponent = () => (
...
-                    <li>Campaigns have been renamed to Batch Changes!</li>
+                    <li>Batch Changes now supports SSH cloned repos. Users can configure SSH access in settings.</li>
                 </ul>
                 <ul className="text-muted mb-0 pl-3">
-                    <li>Batch Changes now supports SSH cloned repos. Users can configure SSH access in settings.</li>
+                    <li>
+                        Burndown charts have been improved: changeset progress is now shown with greater resolution
+                        across the entire lifespan of the batch change.
+                    </li>
                 </ul>
```
Comment (eseliger) [inline client/web/src/enterprise/batches/list/BatchChangesListIntro.tsx:55]: Oh this might have been my mistake but do we actually need two lists here?

**Review by eseliger — APPROVED** [review]
Thanks! 🙏

File: client/web/src/enterprise/batches/list/BatchChangesListIntro.tsx (hunk 2 above)
Comment (LawnGnome) [reply client/web/src/enterprise/batches/list/BatchChangesListIntro.tsx:55]: We shouldn't, no. Previous versions have just been a single list.

File: client/web/src/enterprise/batches/list/BatchChangesListIntro.tsx (hunk 1 above)
Comment (mrnugget) [reply client/web/src/enterprise/batches/list/BatchChangesListIntro.tsx:77]: cc @malomarrec

**Review by mrnugget — APPROVED** [review]

**Review by chrispine — COMMENTED** [review]
LGTM!

**Pull request merged**